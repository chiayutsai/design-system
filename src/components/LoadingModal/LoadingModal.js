import classnames from 'classnames'
import useStyles from 'isomorphic-style-loader/useStyles'
import PropTypes from 'prop-types'
import ModalWrapper from '../ModalWrapper'
import Preloader from '../Preloader'
import styles from './LoadingModal.scss'

const LoadingModal = ({ className }) => {
  useStyles(styles)

  return (
    <ModalWrapper
      disableBodyScroll
      className={classnames(styles.root, className)}>
      <Preloader />
    </ModalWrapper>
  )
}

LoadingModal.propTypes = {
  className: PropTypes.string,
}
LoadingModal.defaultProps = {
  className: '',
}

export default LoadingModal
