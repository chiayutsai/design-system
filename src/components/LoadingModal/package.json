{
  "name": "LoadingModal",
  "version": "0.0.0",
  "private": true,
  "main": "./LoadingModal"
}
