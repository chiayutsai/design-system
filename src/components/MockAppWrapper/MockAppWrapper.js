import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
