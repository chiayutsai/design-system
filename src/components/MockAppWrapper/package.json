{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
