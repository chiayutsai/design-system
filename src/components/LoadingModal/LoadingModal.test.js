import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import LoadingModal from './LoadingModal'

describe('LoadingModal', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <LoadingModal />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
