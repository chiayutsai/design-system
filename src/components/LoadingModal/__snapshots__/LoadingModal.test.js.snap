// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal render correctly 1`] = `
<div
  className="root root"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="preloader"
  >
    <div
      className="preloader_spiner"
    >
      <div
        className="preloader_circles"
      >
        <div
          className="preloader_circle-1"
        />
        <div
          className="preloader_circle-2"
        />
      </div>
    </div>
  </div>
</div>
`;
